### ISSUE TITLE
Training custom data - metrics drop sharply while training

### ISSUE THREAD

**Issue by itaitu** [eriklindernoren/PyTorch-YOLOv3#509]
Hi,
I'm training on a custom dataset with a single class. The dataset size is 1,300 images. 
While training, all of the metrics (loss, accuracy, recall) from all of the Yolo layers, seem to drop sharply and then restart again. Below are examples from Tensorboard:
Loss:
![image](https://user-images.githubusercontent.com/45493794/82737661-47b69200-9d3b-11ea-9ed8-4edc5f60153a.png)

Accuracy:
![image](https://user-images.githubusercontent.com/45493794/82737668-5d2bbc00-9d3b-11ea-933c-479e0509b443.png)

Recall (0.75):
![image](https://user-images.githubusercontent.com/45493794/82737681-70d72280-9d3b-11ea-84c2-735c7ac226c7.png)

Notice that there's only a single drop in these plots, but it's not the first time that i observed this phenomena. In a previous training it happened similarly, a few times.

Attached is also a .txt file of the precision data.
[precision.txt](https://github.com/eriklindernoren/PyTorch-YOLOv3/files/4672321/precision.txt)

Training on a Ubuntu18.04 with GTX-2080i, 16 GB RAM.
Config (mostly defualt values):
[net]
batch=64
subdivisions=8
width=416
height=416
channels=3
momentum=0.9
decay=0.0005
angle=0
saturation = 1.5
exposure = 1.5
hue=.1

learning_rate=0.001
burn_in=1000
max_batches = 500200
policy=steps
steps=400000,450000
scales=.1,.1

Would appreciate any insight to the reasons behind this phenomena.

**Comment by voodoopotato** [comment]
My insight for this is that your network fell into a local minimal while descending the gradients, and due to the momentum was able to get itself out of there before getting stuck.

**Comment by Flova** [comment]
Is this issue still relevant/occurring?

**Comment by itaitu** [comment]
Hi,
We didn't investigate the issue further, as we eventually used the YOLO-v4 implementation in [Darknet](https://github.com/AlexeyAB/darknet) in our project, mainly because of its faster inference times. The issue didn't occur again there.
